### DISCUSSION TITLE
Is there a way to disable auto upload of source maps?

### DISCUSSION THREAD

**Discussion by jpinto321** [getsentry/sentry-react-native#3319] [Q&A] [answered]
Is there like a flag or property or something of sorts, that allows us to disable the auto upload of sourcemaps?

I noticed that sometimes I end up doing a production build, just to test locally, and it always uploads the sourcemaps, which overrides the one in the Sentry server, and if I am testing some code changes, then any future error logs will show with an incorrect stack error, because it doesn't match the release in production.

**Comment by krystofwoldrich** [comment]
Hi, thank you for the question,
we currently don't have a flag to disable Sentry auto upload. There is PR in progress with that feature.

At the moment you can achieve similar behavior by removing your auth token and allow Sentry to fail as described in https://docs.sentry.io/platforms/react-native/troubleshooting/#failing-build-due-to-se

**Comment by tomimarkus991** [comment]
any updates?

**Reply by krystofwoldrich** [reply, marked as answer]
Yes, 

`export SENTRY_DISABLE_AUTO_UPLOAD=true
` is now available for both RN on iOS and Android to quickly disable auto uploads of any debug files including source maps to Sentry.

https://docs.sentry.io/platforms/react-native/sourcemaps/#disable-automatic-upload

**Reply by jenskuhrjorgensen** [reply]
Hi @krystofwoldrich 

If you are using Groovy to configure the Sentry Gradle Integration, what Groovy parameter should I use to achieve the same as the Bash parameter `SENTRY_DISABLE_AUTO_UPLOAD`?

From your [documentation](https://docs.sentry.io/platforms/react-native/manual-setup/manual-setup/#enable-gradle-integration) this is not really clear to me:
![Screenshot 2025-04-28 at 09 06 03](https://github.com/user-attachments/assets/d11729de-c8a1-4f06-8d22-416962c6207a)
vs
![Screenshot 2025-04-28 at 09 06 09](https://github.com/user-attachments/assets/2906e32d-61d9-4fe0-ba59-917f66bb9f22)

**Reply by jenskuhrjorgensen** [reply]
Or am I misinterpreting the code block switch between Bash/Groovy? Normally, it means you can configure something either in this language or in this language, but I guess what you have in Bash in this example does not do the same as what you have in Groovy in this example, so maybe it is more to be read as: "You have the option to configure those couple of settings via Bash on those couple of settings via Groovy"?

**Comment by Traviskn** [comment]
In case anyone is trying to run a local build and is having a hard time getting xcode to pick up on this environment variable, you can add it to the "Bundle React Native code and images" build phase script which you can find by selecting your build target -> build phases
<img width="987" height="487" alt="Screenshot 2025-12-15 at 7 12 08 PM" src="https://github.com/user-attachments/assets/e3069d68-e101-4eab-b2cd-f6938f8bba72" />
